src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx

```diff
@@ -169,9 +169,16 @@ export default class ReviewCollapsibleChapter extends React.Component {
                         page.index,
                       )
                     }
-                    onChange={formData =>
-                      this.onChange(formData, page.arrayPath, page.index)
-                    }
+                    onChange={formData => {
+                      this.onChange(
+                        (page.updateDataHooks || []).reduce(
```

Comment:
What made you choose to make `updateDataHooks` an array, rather than a function? Presumably if someone using the form has several of them they could write their own reduce logic like this.

Reply:
Another related thing: I don't think `hooks` is the right term to use here. It could be confused with React hooks and it also seems like a description of what the form system is providing, rather than what the user is providing. The user is providing an array of update functions, or a callback for a `setData` hook. I wonder if a name like `onSetData` or `onChange` makes more sense and helps a user understand when these are being executed.

Reply:
**`updateDataHooks` as an Array**
I decided to go with an array in case it made sense at any time to memoize any of the functions. I'm not sure if that really makes sense anymore, though, since any memoized function will have to have the same bit of `oldData` and `newData` to benefit from it.

Honestly, I'm not sure if it makes sense as an array. It does give the engineer a perhaps cleaner way to separate any concerns, though. Like, we can write two functions to deal with two separate fields on a page. But like you say, that's still pretty easy to reduce on our own.

**Naming**
...is hard. I'm definitely not set on `updateDataHooks`. I think I meant it to be a temporary name that I never asked alternatives for. You make a good point about the confusion it could cause.

As far as alternatives go, `onSetData` is maybe misleading since this happens _before_ the `setData` call. `onChange` has a lot of overlap with what other components use, while still not being super descriptive in the page config.

:thinking: I'm not sure I have any better suggestions, though. I'll definitely keep pondering this one.